phpBB/posting.php

```diff
@@ -44,6 +41,75 @@
 $submit = $request->is_set_post('post') && !$refresh && !$preview;
 $mode		= $request->variable('mode', '');
 
+// We need to choose which of the url parameters we must trust and ignore the others
+	$forum_id = 0;
+	$topic_id = 0;
+	$post_id = 0;
+
+switch ($mode)
+{
+	case 'popup':
+	case 'smilies':
+		$forum_id = $request->variable('f', 0);
+		break;
+
+	case 'post':
+		$forum_id = $request->variable('f', 0);
+		if (!$forum_id)
+		{
+			$user->setup('posting');
```

Comment:
`NO_FORUM` is defined in common.php and therefore there is no need for `$user->setup('posting');` here.

Reply:
OK